metal/drivers/sifive_buserror0.h

```diff
@@ -0,0 +1,121 @@
+/* Copyright 2020 SiFive, Inc */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#ifndef METAL__DRIVERS__SIFIVE_BUSERROR0_H
+#define METAL__DRIVERS__SIFIVE_BUSERROR0_H
+
+/*!
+ * @file sifive_buserror0.h
+ *
+ * @brief API for configuring the SiFive Bus Error Unit
+ */
+
+#include <metal/compiler.h>
+
+/*!
+ * @brief The set of possible events handled by a SiFive Bus Error Unit
+ */
+typedef enum {
+    METAL_BUSERROR_EVENT_NONE = 0,
+    __METAL_BUSERROR_EVENT_MIN = METAL_BUSERROR_EVENT_NONE,
+    METAL_BUSERROR_EVENT_INST_CORRECTABLE_ECC_ERROR = 2,
+    METAL_BUSERROR_EVENT_INST_UNCORRECTABLE_ECC_ERROR = 3,
+    METAL_BUSERRROR_EVENT_LOAD_STORE_ERROR = 5,
+    METAL_BUSERROR_EVENT_DATA_CORRECTABLE_ECC_ERROR = 6,
+    METAL_BUSERROR_EVENT_DATA_UNCORRECTABLE_ECC_ERROR = 7,
+    __METAL_BUSERROR_EVENT_MAX = METAL_BUSERROR_EVENT_DATA_UNCORRECTABLE_ECC_ERROR,
+} metal_buserror_event_t;
+
+#define METAL_BUSERROR_LOCAL_INTERRUPT_ID 128U
+
+/*!
+ * @brief The handle for a bus error unit
+ */
+struct metal_buserror {};
+
+/*!
+ * @brief Enable or disable the platform interrupt
+ *
+ * @param beu The bus error unit handle
+ * @param event The error event which would trigger the interrupt
+ * @param enabled True if the interrupt should be enabled
+ * @return 0 upon success
+ */
+int metal_buserror_set_platform_interrupt(struct metal_buserror *beu, metal_buserror_event_t event, bool enabled);
+
+/*!
+ * @brief Enable or disable the hart-local interrupt
+ *
+ * @param beu The bus error unit handle
+ * @param event The error event which would trigger the interrupt
+ * @param enabled True if the interrupt should be enabled
+ * @return 0 upon success
+ */
+int metal_buserror_set_local_interrupt(struct metal_buserror *beu, metal_buserror_event_t event, bool enabled);
+
+/*!
+ * @brief Get the error event which caused the most recent interrupt
+ *
+ * This method should be called from within the interrupt handler for the bus error unit interrupt
+ *
+ * @param beu The bus error unit handle
+ * @return The event which caused the interrupt
+ */
+metal_buserror_event_t metal_buserror_get_cause(struct metal_buserror *beu);
+
+/*!
+ * @brief Clear the cause register for the bus error unit
+ *
+ * This method should be called from within the interrupt handler for the bus error unit to un-latch
+ * the cause register for the next event
+ *
+ * @param beu The bus error unit handle
+ * @return 0 upon success
+ */
+int metal_buserror_clear_cause(struct metal_buserror *beu);
```

Comment:
same as above

Reply:
Writing 0 to the cause register should be safe outside of an ISR.